pkg/kubectl/cmd/apply.go

```diff
@@ -240,6 +240,18 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 			visitedNamespaces.Insert(info.Namespace)
 		}
 
+		// Add change-cause annotation to resource info if it should be recorded
+		if cmdutil.ShouldRecord(cmd, info) {
+			recordInObj := info.VersionedObject
+			if info.VersionedObject != nil {
+				recordInObj = info.Object
+			}
+			if err := cmdutil.RecordChangeCause(recordInObj, f.Command(cmd, false)); err != nil {
```

Comment:
This only touches `info`. The change isn't sent to apiserver until `patcher.patch` (L317, to patch the given resource) or `createAndRefresh` (L275, to create a resource)